pilot/pkg/model/push_context.go

```diff
@@ -1442,7 +1443,14 @@ func (ps *PushContext) initServiceAccounts(env *Environment, services []*Service
 				if len(svc.ServiceAccounts) > 0 {
 					accounts = accounts.Copy().InsertAll(svc.ServiceAccounts...)
 				}
-				sa := sets.SortedList(spiffe.ExpandWithTrustDomains(accounts, ps.Mesh.TrustDomainAliases))
+				var sa []string
+				// Skip external services as service accounts for them (SANs in Service Entry)
+				// may not be in SPIFFE format.
+				if svc.Attributes.ServiceRegistry == provider.External {
+					sa = svc.ServiceAccounts
```

Comment:
I have a feeling this will break some valid use cases. The list may be a mix of spiffe, KSAs and even DNS SANs.

I would check each name in the list for dots - KSA don't have dots.